### ISSUE TITLE
Change of blockDeviceMappings produce FailedInflightCheck

### ISSUE THREAD

**Issue by runningman84** [aws/karpenter#3076]
### Version

__Karpenter Version:__ v0.20.0

__Kubernetes Version:__ v1.23.13

### Expected Behavior

A change to the block device mapping in the awsnodetemplate should triggger a node replacement.

### Actual Behavior

The nodes do not change but produce error messages like karpenter  Expected 50Gi of resource ephemeral-storage, but found 20959212Ki (40.0% of expected)

### Steps to Reproduce the Problem

Change AWSNodetemplate

### Resource Specs and Logs

Warning  FailedInflightCheck  13m (x21 over 3h33m)   karpenter  Expected 50Gi of resource ephemeral-storage, but found 20959212Ki (40.0% of expected)

`````
apiVersion: karpenter.k8s.aws/v1alpha1
kind: AWSNodeTemplate
metadata:
  creationTimestamp: "2022-12-19T17:06:35Z"
  generation: 4
  labels:
    kustomize.toolkit.fluxcd.io/name: karpenter-crds
    kustomize.toolkit.fluxcd.io/namespace: flux-system
  name: default
  resourceVersion: "593539611"
  uid: ae67d38e-f041-46f7-94ae-a05e42560653
spec:
  amiFamily: AL2
  blockDeviceMappings:
  - deviceName: /dev/xvda
    ebs:
      deleteOnTermination: true
      encrypted: true
      volumeSize: 50Gi
      volumeType: gp3
  securityGroupSelector:
    aws:eks:cluster-name: mycluster
  subnetSelector:
    kubernetes.io/role/internal-elb: "1"
`````

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by vazkarvishal** [comment]
> The ability to change the current nodes based on changes to the AWSNodeTemplate is tied to node drift can captured in this issue (#1738). We are planning to get to the point where we can do node replacement based on AWSNodeTemplate changes (currently we only support AMI drift). For now, if you wanted your nodes to roll based on a change to your NodeTemplate, you could manually roll the nodes by deleting them manually or enable `ttlSecondsAfterExpired` to roll the nodes after some fixed time.
> 
> @njtran has been owning the design and implementation on this one, feel free to reach out to him if you have specific questions.
> 
> The inflight check is failing right now because we don't do a point-in-time check for the instance type expected resources, it's based on the current state of your Provisioner/NodeTemplate at the time we do the inflight check. Since you updated the available ephemeral-storage, we think that the node should have more storage than it actually does.
> 
> This issue is being addressed in #2944 where we are adding a `.status.allocatable` field to Machines that we launch so that we can get a point-in-time capture of estimated capacity/allocatable

Has this been addressed? We are using Karpenter version 0.22.1 where it is not recycling nodes on change in NodeTemplate (in our case EBS volume size).

**Comment by jonathan-innis** [comment]
> Has this been addressed

It was closed since we deemed it a non-issue. #3408 should also address this FailedInflightCheck being fired when there is a change in the NodeTemplate
